### PR TITLE
[Feature] Added `Flow` `match_id` attribute for efficient overlapping matches updates

### PR DESCRIPTION
**Pull request by viniarck**
### Release notes

- Added ``match_id`` attribute on ``Flow`` as a unique match identifier for efficient overlapping matches updates, minimizing extra DB lookups that would be needed otherwise.
- Match OF10 ``as_dict`` to only included explicit set values.

This picture illustrates a `POST /api/kytos/flow_manager/v2/flows/` on `flow_manager` (that's in progress on https://github.com/kytos-ng/flow_manager/issues/75), by leveraging this `match_id` there's a single update, minimizing DB round trips when inserting or updating a flow, minimizing locks and delegating as much to Mongo whenever applicable

fmanager_mongo

I wasn't going to support OF10 either on this attribute, since we have mapped the issues to deprecate it, but since we haven't yet and the fix was simple I committed it. We can drop OF1.0 on issue #50 .

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -192,6 +192,10 @@ def on_handshake_completed_request_flow_list(self, event):
         if switch.is_enabled():
             self._request_flow_list(switch)
 
+    def handle_handshake_completed_request_flow_list(self, switch):
+        """Request an flow list right after the handshake is completed."""
+        self._request_flow_list(switch)
+
```
Comment (italovalcy) [inline main.py:198]: I believe this should be part of `on_handshake_completed_request_flow_list`, shouldn't it?

**Review by italovalcy — COMMENTED** [review]
Very nice the idea of creating this attribute to work as an index for efficient detect overlapping matches! It will be very useful. Just left a comment regarding the usage of handle function (I believe it is missing something)

**Review by ajoaoff — APPROVED** [review]

File: main.py (hunk 1 above)
Comment (viniarck) [reply main.py]: You're right @italovalcy, I extracted this method but then forgot to call it where it was originally supposed to be, done here https://github.com/kytos-ng/of_core/pull/60/commits/902877f2068897dda54bb130a14318ab87452ff2.

**Comment by viniarck** [comment]
Thanks for reviewing @ajoaoff @italovalcy

**Review by italovalcy — APPROVED** [review]

**Pull request merged**